src/core/src/pass/serialize.cpp

```diff
@@ -1229,21 +1229,27 @@ bool pass::Serialize::run_on_model(const std::shared_ptr<ov::Model>& model) {
         if (xmlDir != m_xmlPath)
             ov::util::create_directory_recursive(xmlDir);
 
-        std::ofstream bin_file(m_binPath, std::ios::out | std::ios::binary);
-        OPENVINO_ASSERT(bin_file, "Can't open bin file: \"" + m_binPath + "\"");
-
-        // create xml file
-        std::ofstream xml_file(m_xmlPath, std::ios::out);
-        OPENVINO_ASSERT(xml_file, "Can't open xml file: \"" + m_xmlPath + "\"");
-
         try {
+            // create bin file
+            std::ofstream bin_file(m_binPath, std::ios::out | std::ios::binary);
+            bin_file.exceptions(std::ofstream::failbit | std::ofstream::badbit);
+            OPENVINO_ASSERT(bin_file, "Can't open bin file: \"" + m_binPath + "\"");
+        
+            // create xml file 
+            std::ofstream xml_file(m_xmlPath, std::ios::out);
+            xml_file.exceptions(std::ofstream::failbit | std::ofstream::badbit);
+            OPENVINO_ASSERT(xml_file, "Can't open xml file: \"" + m_xmlPath + "\"");
```

Comment:
May we return it back in order to not catch asserts and rethrow? It looks like extra behavior

Reply:
Sure! It doesn't seem fine to give additional errors over ```OPENVINO``` assertion errors, I've dropped the catch block for ```std::ofstream::failure```.